arch/arc/core/arc_smp.c

```diff
@@ -44,15 +44,15 @@ u64_t z_arch_smp_switch_in_isr(void)
 	u32_t new_thread;
 	u32_t old_thread;
 
-	if (!_current_cpu->swap_ok) {
-		return 0;
-	}
-
 	old_thread = (u32_t)_current;
 
 	new_thread = (u32_t)z_get_next_ready_thread();
 
 	if (new_thread != old_thread) {
+/* should check whether new_thread to relace old_thread */
+#ifdef CONFIG_TIMESLICING
+		z_reset_time_slice();
+#endif
```

Comment:
This code too looks like it's trying to duplicate logic that should be portable in the scheduler.  Actually lots of code around here does, predating this PR.  I think there may be a deeper problem here we may need to look at about why this had to be copied.